### PR TITLE
fix: revert semantic version changes since it cause build fail

### PR DESCRIPTION
**Pull request by shruthirai**

### CODE REVIEW

**Comment by hkfb** [comment]
The build seems to fail because of this:

`[semantic-release]: node version >=18 is required. Found v14.21.2.`

Could we force a newer node version in the pipeline instead?

**Comment by shruthirai** [comment]
> The build seems to fail because of this:
> 
> `[semantic-release]: node version >=18 is required. Found v14.21.2.`
> 
> Could we force a newer node version in the pipeline instead?

Yes, will do that

**Comment by shruthirai** [comment]
@hkfb Now I have used volta and pinned the node version to 18.0.0 but I can also using standard method using engines ([engines](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) and set the node version >=18, if that's recommended

**Comment by hkfb** [comment]
> @hkfb Now I have used volta and pinned the node version to 18.0.0 but I can also using standard method using engines ([engines](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) and set the node version >=18, if that's recommended

Looks like it's still using v14 in the workflow, so not sure if this works

```
Run actions/setup-node@v2
  with:
    node-version: 14.x
    registry-url: https://registry.npmjs.org
    always-auth: false
    check-latest: false
    token: ***
Found in cache @ /opt/hostedtoolcache/node/14.[2](https://github.com/emerson-eps/color-tables/actions/runs/4042464412/jobs/6950198460#step:3:2)1.2/x6[4](https://github.com/emerson-eps/color-tables/actions/runs/4042464412/jobs/6950198460#step:3:4)
```

Is there a way to set it up in the github workflow?

**Comment by shruthirai** [comment]
> > @hkfb Now I have used volta and pinned the node version to 18.0.0 but I can also using standard method using engines ([engines](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) and set the node version >=18, if that's recommended
> 
> Looks like it's still using v14 in the workflow, so not sure if this works
> 
> ```
> Run actions/setup-node@v2
>   with:
>     node-version: 14.x
>     registry-url: https://registry.npmjs.org
>     always-auth: false
>     check-latest: false
>     token: ***
> Found in cache @ /opt/hostedtoolcache/node/14.[2](https://github.com/emerson-eps/color-tables/actions/runs/4042464412/jobs/6950198460#step:3:2)1.2/x6[4](https://github.com/emerson-eps/color-tables/actions/runs/4042464412/jobs/6950198460#step:3:4)
> ```
> 
> Is there a way to set it up in the github workflow?
> 
> Edit: maybe here:
> 
> https://github.com/emerson-eps/color-tables/blob/bd762994ffa4861938037dea0865d80020864d2b/.github/workflows/node.js.yml#L23

Let me try using "engines" in package file and see if that helps

**Comment by shruthirai** [comment]
@hkfb Not sure why its failing though I have version 18.0.0 but its failing saying 
npm ERR! notsup Required: {"node":">=18.0"}
npm ERR! notsup Actual:   {"npm":"8.1[9](https://github.com/emerson-eps/color-tables/actions/runs/4043966272/jobs/6953535089#step:4:10).3","node":"v16.19.0"}

Tried npm ci as well.

**Comment by hkfb** [comment]
> @hkfb Not sure why its failing though I have version 18.0.0 but its failing saying npm ERR! notsup Required: {"node":">=18.0"} npm ERR! notsup Actual: {"npm":"8.1[9](https://github.com/emerson-eps/color-tables/actions/runs/4043966272/jobs/6953535089#step:4:10).3","node":"v16.19.0"}
> 
> Tried npm ci as well.

The problem is that you restrict the node version in package.json to v18, while in the gh-pages workflow it uses the default version (16). It shouldn't be necessary to restrict the version.

File: react-app/package.json [hunk 1]
```diff
@@ -12,6 +12,9 @@
   "files": [
     "dist"
   ],
+  "engines": {
+    "node": ">=18.0"
```
Comment (hkfb) [inline react-app/package.json:16]: This shouldn't be necessary, and causes the gh-pages workflow to fail.

File: react-app/package.json [hunk 2]
```diff
@@ -92,5 +95,8 @@
   "peerDependencies": {
     "react": "^17.0.2",
     "react-dom": "^17.0.2"
+  },
+  "volta": {
+    "node": "18.0.0"
```
Comment (hkfb) [inline react-app/package.json:100]: What is this used for?

File: react-app/package.json (hunk 1 above)
Comment (shruthirai) [reply react-app/package.json]: Removed

File: react-app/package.json (hunk 2 above)
Comment (shruthirai) [reply react-app/package.json]: Not sure why but when I switch node version through nvm it doesn't switch to latest and hence had to switch using volta(https://docs.volta.sh/guide/#why-volta) for switching to 18.0 version but have removed it now.

File: react-app/.npmrc [hunk 3]
```diff
@@ -1 +1,2 @@
-node-options="--unhandled-rejections=strict"
\ No newline at end of file
+node-options="--unhandled-rejections=strict"
+engine-strict=true
```
Comment (hkfb) [inline react-app/.npmrc:2]: Not sure if this is needed?

Comment (shruthirai) [reply react-app/.npmrc:2]: Yes, just noticed, removing it

**Comment by hkfb** [comment]
I guess you also need to bump node in gh-pages pipeline: https://github.com/emerson-eps/color-tables/blob/bd762994ffa4861938037dea0865d80020864d2b/.github/workflows/gh-pages.yml#L18

**Comment by shruthirai** [comment]
> I guess you also need to bump node in gh-pages pipeline:
> 
> https://github.com/emerson-eps/color-tables/blob/bd762994ffa4861938037dea0865d80020864d2b/.github/workflows/gh-pages.yml#L18

It seems like build is passing, so maybe not required

**Review by hkfb — APPROVED** [review]

**Comment by shruthirai** [comment]
It still using old package lock file, will update it and then merge

**Comment by shruthirai** [comment]
> It still using old package lock file, will update it and then merge ( as you see engines >= 18 is still present in lock file)

I have issue in updating package lock file because it still takes old node version even though i switch to 18.0.0. This may be because I have Volta installed for my new project and causing conflict . Tried removing nodejs file from program files and then use nvm as suggested in stackoverflow but dint work. Maybe you can update the lock file from your side or will talk tomorrow.

image

**Pull request merged**